### PR TITLE
Host admins to freeze accounts

### PR DESCRIPTION
**Pull request by Betree**
Resolve https://github.com/opencollective/opencollective/issues/4694
Require https://github.com/opencollective/opencollective-api/pull/7328
Demo: https://www.loom.com/share/6df6e9ddcca843f8b470f7dad78d091f

- Implements the menu + modal to (un)freeze accounts on host admin collective cards
- Show a banner in the frontend

image

### CODE REVIEW

File: components/collective-page/CategoryHeader.js [hunk 1]
```diff
@@ -99,6 +99,7 @@ const getCategoryData = (intl, collective, category) => {
         ),
       };
     case NAVBAR_CATEGORIES.CONTRIBUTE:
+      // TODO: Hide "Become a financial contributor." if can't contribute
```
Comment (Betree) [inline components/collective-page/CategoryHeader.js:102]: That will be for https://github.com/opencollective/opencollective-frontend/pull/7628

**Review by SudharakaP — COMMENTED** [review]
@Betree : One thing I noticed is that a user can directly go to `/contribute` url of a frozen collective and access the contribution flow. I think we should ideally block this as well as otherwise if a collective `/contribute` section is linked somewhere and then the host freezes it that doesn't block users who have the direct link from contributing. 🤔

**Comment by Betree** [comment]
> @Betree : One thing I noticed is that a user can directly go to `/contribute` url of a frozen collective and access the contribution flow. I think we should ideally block this as well as otherwise if a collective `/contribute` section is linked somewhere and then the host freezes it that doesn't block users who have the direct link from contributing. thinking

That's addressed in https://github.com/opencollective/opencollective-frontend/pull/7628, which is about adapting all the pages for frozen accounts.

File: components/GlobalWarnings.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    );
+  } else if (LoggedInUser && LoggedInUser.isLimited) {
+    // Limited user accounts
+    return (
+      <GlobalWarningContainer>
+        <FormattedMessage
+          id="warning.limitedAccount"
+          defaultMessage="Your account is currently limited. If you think this is a mistake, please contact support@opencollective.com."
+        />
+      </GlobalWarningContainer>
```
[the remark this answers is not in the document]
Comment (Betree) [reply components/GlobalWarnings.js:47]: Yes, it's not a new feature. If you look below, this component is a rename of [`UserWarnings`](https://github.com/opencollective/opencollective-frontend/pull/7617/files#diff-15589f48ab67a68706758870d0c27b3bea8385bd8dc30a02af48aad6ca0b6b09) on which I've added support for showing the banner for frozen accounts

File: components/host-dashboard/FreezeAccountModal.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+              fontWeight="700"
+              lineHeight="24px"
+              color="black.800.900"
+              mb="6px"
+              htmlFor="freeze-account-message"
+            >
+              <FormattedMessage defaultMessage="Include a message to the Collective admins (Optional)" />
+              <br />
+              <Span fontSize="13px" fontWeight="400">
+                <FormattedMessage defaultMessage="They will also be notified of this unfreeze via auto-email notification." />
```
[the remark this answers is not in the document]
Comment (Betree) [reply components/host-dashboard/FreezeAccountModal.js:66]: I've applied the [wording from Kayla](https://github.com/opencollective/opencollective/issues/4694#issuecomment-1075765900) in there. We need to discuss these things on the issue (not the PR) to make sure that the relevant people can participate. Nobody corrected what she proposed in there and I'm personally ok with this string so I'll go with it, but feel free to comment on the issue if you feel that we should discuss that one.

[the remark this answers is not in the document]
Comment (SudharakaP) [reply components/host-dashboard/FreezeAccountModal.js:66]: @Betree : So Kayla replied at, https://github.com/opencollective/opencollective/issues/4694#issuecomment-1083385610. We can use either, "They will also be notified of this unfreeze via auto-email." or my suggestion above.

**Pull request merged**